tcn_hpl/data/utils/ptg_datagenerator.py

```diff
@@ -122,22 +125,51 @@ def filter_dset_by_split(split):
         activity_config = yaml.safe_load(stream)
     activity_labels = activity_config["labels"]
 
+    activity_labels_desc_mapping = {}
     with open(f"{output_data_dir}/mapping.txt", "w") as mapping:
         for label in activity_labels:
             i = label["id"]
             label_str = label["label"]
+            if "description" in label.keys():
+                # activity_labels_desc_mapping[label["description"]] = label["label"]
+                activity_labels_desc_mapping[label["description"]] = label["id"]
+            elif "full_str" in label.keys():
+                # activity_labels_desc_mapping[label["full_str"]] = label["label"]
+                activity_labels_desc_mapping[label["full_str"]] = label["id"]
             if label_str == "done":
                 continue
             mapping.write(f"{i} {label_str}\n")
-
+    
+    print(f"label mapping: {activity_labels_desc_mapping}")
+    # exit()
+    dset = kwcoco.CocoDataset(config["data_gen"]["dataset_kwcoco"])
+    # Check if the dest has activity gt, if it doesn't then add it
+    if not "activity_gt" in list(dset.imgs.values())[0].keys():
+        print("adding activity ground truth to the dataset")
+        from angel_system.data.common.kwcoco_utils import add_activity_gt_to_kwcoco
```

Comment:
Another in-line import that likely ought to be at the top if possible, otherwise document why this is here.

Reply:
This exists in the current main branch. 

I can move it to the top.